UI/Sources/ShopFinder/Views/ShopMapView.swift

```diff
@@ -41,122 +35,105 @@ extension View {
-            interactionModes: .all,
-            showsUserLocation: true,
-                userTrackingMode: .constant(.none),
-                annotationItems: [self.shopLocation]) { place in
-                MapAnnotation(coordinate: place.shop.location.coordinate, anchorPoint: CGPoint(x: 0.5, y: 1)) {
-                    ShopAnnotationView(shopLocation: place)
-                        .shadow(color: .gray, radius: 3, x: 2, y: 2)
+        Map(position: $position) {
+            Annotation("", coordinate: shop.location.coordinate, anchor: .bottom) {
+                VStack(spacing: -2) {
+                    HStack {
+                        Button(action: {
+                            showingAlert.toggle()
+                        }) {
+                            Image(systemName: "car.fill")
+                                .resizable()
+                                .scaledToFit()
+                                .frame(width: 24, height: 24)
+                                .padding(8)
+                                .background(Color.accent())
+                                .foregroundColor(.onAccent())
+                                .cornerRadius(4)
+                        }
+                        .navigateToShopAlert(isPresented: $showingAlert, shop: shop)
+
+                        VStack(alignment: .leading, spacing: 0) {
+                            AddressView(provider: shop)
+                        }
+                        .foregroundColor(.primary)
+                    }
+                    .padding([.leading, .top, .bottom], 4)
+                    .padding(.trailing, 8)
+                    .background(Color.systemBackground)
+                    .cornerRadius(8)
+                    .onTapGesture {
+                        showingDetails.toggle()
+                    }
+                    
+                    Image(systemName: "arrowtriangle.down.fill")
+                        .foregroundColor(.systemBackground)
+                    
+                    Spacer(minLength: 16)
                 }
+                .compositingGroup()
+                .opacity(showingDetails ? 1 : 0)
             }
-        } else {
-            Map(coordinateRegion: .init(get: { region }, set: { _ in }),
-            interactionModes: .all,
-            showsUserLocation: true,
-            userTrackingMode: .constant(.follow),
-                annotationItems: [self.shopLocation]) { place in
-                MapAnnotation(coordinate: place.shop.location.coordinate, anchorPoint: CGPoint(x: 0.5, y: 1)) {
-                    ShopAnnotationView(shopLocation: place)
-                        .shadow(color: .gray, radius: 3, x: 2, y: 2)
-                }
+            .annotationTitles(.hidden)
+            .annotationSubtitles(.hidden)
+            Annotation(shop.name, coordinate: shop.location.coordinate) {
+                shopAnnotation
             }
-
+            UserAnnotation()
+        }
+        .task {
```

Comment:
Muss hier, in diesem Use Case, nicht eher `.onAppear {}` verwendet werden?

Reply:
Warum würdest du `onAppear` nutzen? Es soll dich nur einmalig laufen, da wir initial mit `.automatic` starten und so wird es auf den zu berechnenden Wert von `mode` gesetzt.

Reply:
Wie verhält sich die View, wenn ich z.B. zum Profil navigieren und später auf die Map zurück. Wird dann die User Position aktualisiert?

Reply:
Sollte, da dies die `UserAnnotation` macht. Der Task ändert nur die `position` zu dem aktuellen `mode`. Falls das von dir beschriebene nicht funktioniert müssten wir die `UserAnnoation` anpassen.

Reply:
Hast du das Verhalten ausprobiert?

Reply:
Stimmt, funktioniert wie erwartet. Bin zur nächsten Ecke gelaufen und alles passt.